### PR TITLE
Add support for 0-dim requests via REST + more scalar tests

### PR DESCRIPTION
**Pull request by dkalinowski**

### CODE REVIEW

File: src/test/prediction_service_test.cpp [hunk 1]
```diff
@@ -880,6 +880,27 @@ TYPED_TEST(TestPredict, SuccesfullInferenceOnModelWithScalar) {
     this->checkOutputShape(response, {}, SCALAR_MODEL_OUTPUT_NAME);
 }
 
+TYPED_TEST(TestPredict, Succesfull0DimInferenceOnModelWithDynamicBatch) {
```
Comment (dkalinowski) [inline src/test/prediction_service_test.cpp:883]: Missing test from previous PR

File: src/test/rest_utils_test.cpp [hunk 2]
```diff
@@ -103,6 +103,15 @@ TEST_F(TFSMakeJsonFromPredictResponseStringTest, PositiveColumn) {
     EXPECT_EQ(json, expected_json);
 }
 
+TEST_F(TFSMakeJsonFromPredictResponseStringTest, ScalarColumn) {
```
Comment (dkalinowski) [inline src/test/rest_utils_test.cpp:106]: Missing scalar tests from previous PRs

File: src/test/kfs_rest_parser_test.cpp [hunk 3]
```diff
@@ -297,6 +297,31 @@ TEST_F(KFSRestParserTest, parseValidRequestBOOL) {
...
+    })";
+    auto status = parser.parse(request.c_str());
+    ASSERT_EQ(status, StatusCode::OK);
+
+    auto proto = parser.getProto();
+    ASSERT_EQ(proto.inputs_size(), 1);
+    ASSERT_EQ(proto.inputs()[0].name(), "input0");
+    ASSERT_THAT(proto.inputs()[0].shape(), ElementsAre(10, 224, 0, 3));
+    ASSERT_EQ(proto.inputs()[0].datatype(), "INT64");
+    ASSERT_EQ(proto.inputs()[0].contents().fp32_contents_size(), 0);
```
Comment (michalkulakowski) [inline src/test/kfs_rest_parser_test.cpp:319]: We should check _content_size for every datatype here. Please add comment with explanation why checking others than int64_contents_size makes sense

File: src/test/tfs_rest_parser_row_test.cpp [hunk 4]
```diff
@@ -136,6 +163,20 @@ TEST(TFSRestParserRow, ValidShape_1x2) {
     EXPECT_THAT(asVector<float>(parser.getProto().inputs().at("i").tensor_content()), ElementsAre(155.0, 56.0));
 }
 
+TEST(TFSRestParserRow, ValidShape_2x0) {
+    TFSRestParser parser(prepareTensors({{"i", {2, 1}}}, ovms::Precision::I32));
```
Comment (dkalinowski) [inline src/test/tfs_rest_parser_row_test.cpp:167]:
```suggestion
    TFSRestParser parser(prepareTensors({{"i", {2, 0}}}, ovms::Precision::I32));
```

**Review by michalkulakowski — APPROVED** [review]

File: src/rest_parser.cpp [hunk 5]
```diff
@@ -177,8 +177,25 @@ bool TFSRestParser::parseInstance(rapidjson::Value& doc) {
...
+            // If previous iterations already increased number of dimensions
+            // it means we have incorrect json
+            if (proto.tensor_shape().dim_size() > 1) {
+                return false;
+            }
+            if (!addValue(proto, itr.value)) {
+                return false;
+            }
+        } else {
+            // If previous iterations already added instances (bateches)
```
Comment (dtrawins) [inline src/rest_parser.cpp:190]:
```suggestion
            // If previous iterations already added instances (batches)
```

**Review by dtrawins — APPROVED** [review]

**Pull request merged**